documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -275,6 +275,20 @@
               description: 'Prometheus %(prometheusName)s has dropped {{ printf "%%.0f" $value }} targets because some samples exceeded the configured label_limit, label_name_length_limit or label_value_length_limit.' % $._config,
             },
           },
+          {
+            alert: 'PrometheusTargetSyncFailure',
+            expr: |||
+              increase(prometheus_target_sync_failed_total{%(prometheusSelector)s}[5m]) > 0
+            ||| % $._config,
+            'for': '5m',
```

Comment:
I am wondering if that is fine.

Service discovery might not trigger every 5 minutes. It might be wrong "one time", then not re-trigger. Should we alert when > 0?


cc @beorn7 what do you think?

Reply:
Wouldn't we just perpetually alert then? `prometheus_target_sync_failed_total` never will decrease, even if the problem has been resolved.

Reply:
Yes, maybe we need a gauge after all? Let's collect more feedback.

Reply:
Yeah, it definitely could help because the operator would know when they have successfully fixed the issue, not only if it ever existed.

Reply:
Not sure, either. My thinking is that for events that clearly happen less often than a scrape, a gauge should be fine. And I guess a sync naturally happens much less often than a scrape (but I might be wrong here). So if we used a gauge, it would contain the number of failures in the last sync of that group.

Reply:
The minimum for syncs is 5 seconds, and in dynamic environments we can reach that cadence (e.g. kubernetes).

Reply:
Hmm, yeah, so I guess the best trade-off has to be done by someone more familiar with the scraping code.

Fundamentally, if an error can happen as often as once every 5s, I'd go for a counter, even if that makes precise alerting a bit harder.

We could extend the time range to an hour or two to make it more likely that the alert keep firing, or at least the alert will fire for at least an hour or two, which is, of course, also a problem because it will continue to fire for an hour or two even if the problem is solved. The latter can be addressed in the description ("In the last two hours, there were x errors."), so that at least the receiver understands why the alert is still firing.

On the other hand, not all alerts have the nice property of only resolving if the problem is solved for good. So we could also just accept that here. (I'd still go at least a bit higher than 5m with the time range.)

Reply:
I think that in this scenario it's especially important to know when the problem has been resolved, so having a potential hour(s) long delay isn't optimal. I definitely think that extending to 15m (or even 30) is warranted though.